src/sentry_core.c

```diff
@@ -192,39 +208,30 @@ sentry_clear_modulecache(void)
     sentry__modulefinder_cleanup();
 }
 
-const sentry_options_t *
-sentry_get_options(void)
-{
-    return g_options;
-}
-
 static void
 set_user_consent(sentry_user_consent_t new_val)
 {
-    sentry__mutex_lock(&g_options_mutex);
-    if (!g_options) {
-        sentry__mutex_unlock(&g_options_mutex);
-        return;
-    }
-    g_options->user_consent = new_val;
-    sentry__mutex_unlock(&g_options_mutex);
-    sentry_path_t *consent_path
-        = sentry__path_join_str(g_options->database_path, "user-consent");
-    switch (new_val) {
-    case SENTRY_USER_CONSENT_GIVEN:
-        sentry__path_write_buffer(consent_path, "1\n", 2);
-        break;
-    case SENTRY_USER_CONSENT_REVOKED:
-        sentry__path_write_buffer(consent_path, "0\n", 2);
-        break;
-    case SENTRY_USER_CONSENT_UNKNOWN:
-        sentry__path_remove(consent_path);
-        break;
-    }
-    sentry__path_free(consent_path);
+    SENTRY_WITH_OPTIONS (options) {
+        sentry__atomic_store((long *)&options->user_consent, new_val);
```

Comment:
_nit_: You could compare the return value with `new_val` and then skip everything below if it didn't change.

```suggestion
        prev_val = sentry__atomic_store((long *)&options->user_consent, new_val);
```